fearless/Modules/Staking/StakingPayoutConfirmation/StakingPayoutConfirmationPresenter.swift

```diff
@@ -139,96 +89,167 @@ extension StakingPayoutConfirmationPresenter: StakingPayoutConfirmationPresenter
+            return
         }
 
-        self.rewardDestination = rewardDestination
+        guard rewardAmount >= fee else {
+            didFailPayout(error: StakingPayoutConfirmError.tinyPayout)
+            return
+        }
 
-        provideViewModel()
+        interactor.submitPayout()
     }
 
-    func didReceive(rewardDestinationError: Error) {
-        handle(error: rewardDestinationError)
-    }
+    func presentAccountOptions(for viewModel: AccountInfoViewModel) {
+        let locale = view?.localizationManager?.selectedLocale ?? Locale.current
 
-    func didStartPayout() {
-        view?.didStartLoading()
+        if let view = view,
+           let chain = WalletAssetId(rawValue: asset.identifier)?.chain {
+            wireframe.presentAccountOptions(
+                from: view,
+                address: viewModel.address,
+                chain: chain,
+                locale: locale
+            )
+        }
     }
+}
 
-    func didCompletePayout(txHash: String) {
-        logger?.info("Did send payouts: \(txHash)")
+// MARK: - StakingPayoutConfirmationInteractorOutputProtocol
 
-        view?.didStopLoading()
+extension StakingPayoutConfirmationPresenter: StakingPayoutConfirmationInteractorOutputProtocol {
+    func didReceiveFee(result: Result<Decimal, Error>) {
+        switch result {
+        case let .success(fee):
+            self.fee = fee
+            provideFee()
 
-        wireframe.complete(from: view)
+        case let .failure(error):
+            logger?.error("Did receive fee error: \(error)")
+        }
     }
 
-    func didFailPayout(error: Error) {
-        view?.didStopLoading()
+    func didReceivePriceData(result: Result<PriceData?, Error>) {
+        switch result {
+        case let .success(priceData):
+            self.priceData = priceData
+            provideFee()
+            provideViewModel()
 
-        handle(error: error)
+        case let .failure(error):
+            logger?.error("Price data subscription error: \(error)")
+        }
     }
 
-    func didReceive(paymentInfo: RuntimeDispatchInfo) {
-        if let feeValue = BigUInt(paymentInfo.fee),
-           let fee = Decimal.fromSubstrateAmount(feeValue, precision: asset.precision) {
-            self.fee = fee
-        } else {
-            fee = nil
+    func didReceiveAccountInfo(result: Result<AccountInfo?, Error>) {
+        switch result {
+        case let .success(accountInfo):
+            if let availableValue = accountInfo?.data.available {
+                balance = Decimal.fromSubstrateAmount(
+                    availableValue,
+                    precision: asset.precision
+                )
+            } else {
+                balance = 0.0
+            }
+
+        case let .failure(error):
+            logger?.error("Account Info subscription error: \(error)")
         }
+    }
 
-        provideFee()
+    func didReceiveStashItem(result: Result<StashItem?, Error>) {
```

Comment:
We just don't need stash in presenter other than to create reward destination. I think it is simpler to store it in interactor as we did for reward destination change module